pkg/internal/checkup/pod/spec.go

```diff
@@ -0,0 +1,133 @@
+	// https://github.com/cri-o/cri-o/blob/fa0fa5de1c17ddd7b6fcdbc030b6b571ce37e643/pkg/annotations/annotations.go
+
+	// CRIOCPULoadBalancingAnnotation indicates that load balancing should be disabled for CPUs used by the container
+	CRIOCPULoadBalancingAnnotation = "cpu-load-balancing.crio.io"
+
+	// CRIOCPUQuotaAnnotation indicates that CPU quota should be disabled for CPUs used by the container
+	CRIOCPUQuotaAnnotation = "cpu-quota.crio.io"
+
+	// CRIOIRQLoadBalancingAnnotation indicates that IRQ load balancing should be disabled for CPUs used by the container
+	CRIOIRQLoadBalancingAnnotation = "irq-load-balancing.crio.io"
+)
+
+const Disable = "disable"
+
+func WithOwnerReference(ownerName, ownerUID string) PodOption {
+	return func(pod *corev1.Pod) {
+		if ownerUID != "" && ownerName != "" {
+			pod.ObjectMeta.OwnerReferences = append(pod.ObjectMeta.OwnerReferences, metav1.OwnerReference{
+				APIVersion: "v1",
+				Kind:       "Pod",
+				Name:       ownerName,
+				UID:        types.UID(ownerUID),
+			})
+		}
+	}
+}
+
+func WithLabels(labels map[string]string) PodOption {
+	return func(pod *corev1.Pod) {
+		if pod.Labels == nil {
+			pod.Labels = map[string]string{}
+		}
+		for key, value := range labels {
+			pod.Labels[key] = value
+		}
+	}
+}
+
+func WithHugepagesVolume() PodOption {
+	return func(pod *corev1.Pod) {
+		pod.Spec.Volumes = append(pod.Spec.Volumes,
+			corev1.Volume{
+				Name: "hugepages",
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{
+						Medium: "HugePages",
+					},
+				},
+			})
+	}
+}
+
+func WithoutCRIOCPULoadBalancing() PodOption {
+	return func(pod *corev1.Pod) {
+		if pod.ObjectMeta.Annotations == nil {
+			pod.ObjectMeta.Annotations = map[string]string{}
+		}
+
+		pod.ObjectMeta.Annotations[CRIOCPULoadBalancingAnnotation] = Disable
+	}
+}
+
+func WithoutCRIOCPUQuota() PodOption {
+	return func(pod *corev1.Pod) {
+		if pod.ObjectMeta.Annotations == nil {
+			pod.ObjectMeta.Annotations = map[string]string{}
+		}
+
+		pod.ObjectMeta.Annotations[CRIOCPUQuotaAnnotation] = Disable
+	}
+}
+
+func WithoutCRIOIRQLoadBalancing() PodOption {
+	return func(pod *corev1.Pod) {
+		if pod.ObjectMeta.Annotations == nil {
+			pod.ObjectMeta.Annotations = map[string]string{}
+		}
+
+		pod.ObjectMeta.Annotations[CRIOIRQLoadBalancingAnnotation] = Disable
+	}
+}
+
+func WithNodeSelector(nodeName string) PodOption {
+	return func(pod *corev1.Pod) {
+		if nodeName == "" {
+			return
+		}
+
+		if pod.Spec.NodeSelector == nil {
+			pod.Spec.NodeSelector = map[string]string{}
+		}
+
+		pod.Spec.NodeSelector[corev1.LabelHostname] = nodeName
+	}
+}
+
+func WithPodContainer(container *corev1.Container) PodOption {
+	return func(pod *corev1.Pod) {
+		pod.Spec.Containers = append(pod.Spec.Containers, *container)
+	}
+}
+
+func NewPod(name string, opts ...PodOption) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PodSpec{},
```

Comment:
Please check if this initialization is necessary.
I think its redundant because Spec is not a pointer.

Reply:
You're right. Removed. thanks!